packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -6,6 +6,68 @@ import '../table_data.dart';
 import '../trees.dart';
 import 'collapsible_mixin.dart';
 
+class FlatTable<T> extends StatefulWidget {
+  const FlatTable({
+    Key key,
+    @required this.columns,
+    @required this.data,
+    @required this.keyFactory,
+  })  : assert(columns != null),
+        assert(keyFactory != null),
+        assert(data != null),
+        super(key: key);
+
+  /// The columns to show in this table.
+  final List<ColumnData<T>> columns;
+
+  /// The tree structure of rows to show in this table.
+  final List<T> data;
+
+  /// Factory that creates keys for each row in this table.
+  ///
+  /// It uses [PageStorageKey] to restore the table's state when re-opening
+  /// its page.
+  final String Function(T data) keyFactory;
+
+  @override
+  _FlatTableState<T> createState() => _FlatTableState<T>();
+}
+
+class _FlatTableState<T> extends NestingTableState<T, FlatTable<T>> {
+  @override
+  bool get startAtBottom => true;
+
+  @override
+  TreeNode<TreeNode> asTreeNode(T node) => null;
```

Comment:
remove this method from the base class rather than creating an oddly nullable api.